Libs/RaptureCore/src/main/java/rapture/dp/DefaultDecisionProcessExecutor.java

```diff
@@ -522,6 +525,39 @@ private void markAsFinished(final WorkOrder workOrder, final Worker worker, Work
             // remove the completed id from the master list
             ids.remove(id);
             workOrder.setPendingIds(ids);
+            
+            // Clear up temporary output storage
+            BlobApi blobApi = Kernel.getBlob();
+            String workOrderURI = worker.getWorkOrderURI();
+            String outputUri = (workOrderURI.startsWith("//")) 
+            		? Scheme.BLOB+":"+workOrderURI
+                	: Scheme.BLOB + (workOrderURI.substring(workOrderURI.indexOf(':')));
+            String auth = new RaptureURI(workOrderURI).toAuthString();
+            
+            if (blobApi.blobRepoExists(kernelUser, auth)) {
+    	        for (String workerId : workOrder.getWorkerIds()) {
+    	            String uri =outputUri+"#"+workerId;
+    	            BlobContainer blob = blobApi.getBlob(kernelUser, uri);
+    	            if (blob != null) {
+    	            	Map<String, String> outputs = workOrder.getOutputs();
+    	            	if (outputs == null) {
+    	            		outputs = new LinkedHashMap<>();
+    	            		workOrder.setOutputs(outputs);
+    	            	}
+    	            	outputs.put(workerId, new String(blob.getContent()));
+    	            	blobApi.deleteBlob(kernelUser, uri);
```

Comment:
Why do we delete these blobs? What are they used by? (i.e. why are they temporary?)